test/unittest/statistics/dds/StatisticsQosTests.cpp

```diff
@@ -63,20 +63,128 @@ class StatisticsFromXMLProfileTests : public ::testing::Test
     {
     public:
 
-        PublisherImpl*  get_publisher_impl()
+        efd::Publisher*  get_publisher()
         {
-            return builtin_publisher_impl_;
+            return builtin_publisher_;
         }
 
     };
 
     eprosima::fastdds::statistics::dds::DomainParticipant* statistics_participant = nullptr;
-    eprosima::fastdds::statistics::dds::PublisherImpl* statistics_publisher_impl = nullptr;
+    eprosima::fastdds::dds::Publisher* statistics_publisher = nullptr;
 
     void test_setup_XMLConfigurationForStatisticsDataWritersQoS(
             eprosima::fastdds::statistics::dds::DomainParticipant*& _statistics_participant,
-            eprosima::fastdds::statistics::dds::PublisherImpl*& _statistics_publisher_impl)
+            eprosima::fastdds::dds::Publisher*& _statistics_publisher)
     {
+        const char* xml =
+                "                                                                                                               \
+            <?xml version=\"1.0\" encoding=\"utf-8\"  ?>                                                                    \
+            <dds xmlns=\"http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles\">                                            \
+            <profiles>                                                                                                      \
+                    <participant profile_name=\"statistics_participant\" is_default_profile=\"true\">                       \
+                    <rtps>                                                                                                  \
+                            <propertiesPolicy>                                                                              \
+                            <properties>                                                                                    \
```

Comment:
This tag should also be indented. It seems like the `8` instead `4` whitespaces indentation is caused because some levels are not correctly indented. Please, review the complete XML definition to fix this issue.